Mathlib/CategoryTheory/Category/RelCat.lean

```diff
@@ -35,4 +39,128 @@ instance rel : LargeCategory RelCat where
   comp f g x z := ∃ y, f x y ∧ g y z
 #align category_theory.rel CategoryTheory.rel
 
+@[ext] theorem rel_ext {X Y : RelCat} (f g : X ⟶ Y) (h : ∀ a b, f a b ↔ g a b) : f = g :=
+  funext₂ (fun a b => propext (h a b))
+
+namespace RelCat.Hom
+
+protected theorem rel_id (X : RelCat) : 𝟙 X = (· = ·) := rfl
+
+protected theorem rel_comp {X Y Z : RelCat} (f : X ⟶ Y) (g : Y ⟶ Z) : f ≫ g = Rel.comp f g := rfl
+
+@[simp] theorem rel_id_apply₂ (X : RelCat) (x y : X) : (𝟙 X) x y ↔ x = y := by
+  rw [RelCat.Hom.rel_id]
+
+@[simp] theorem rel_comp_apply₂ {X Y Z : RelCat} (f : X ⟶ Y) (g : Y ⟶ Z) (x : X) (z : Z) :
```

Comment:
I am not completely sure we want to tag `rel_comp_apply₂` and `rel_id_apply₂` with the `simp` attribute.

Reply:
`types_id_apply` and `types_comp_apply` in `CategoryTheory.Types` are also tagged with simp, is there any notable difference in how these behave?

Reply:
`CategoryTheory.Types` is *ancient*. It's not a very good model for how to do things. In fact, @joelriou (or anyone better versed in the category theory library than me), would you mind reworking `CategoryTheory.Types` to be more "modern"?

Reply:
Still not really sure what issues these simps could cause, but I removed it just in case